### PR TITLE
test: add a test for binary column

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve?
Add a test for the binary columns to make sure that all binary number inside 0x00 to 0xFF is good to encode to canal-json format.

Issue Number: ref #4813

### Check List 

Tests 

 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: tests/integration_tests/canal_json_adapter_compatibility/data/data.sql [hunk 1]
```diff
@@ -123,3 +123,35 @@ CREATE TABLE test_ddl2
...
+    x'000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F202122232425262728292A2B2C2D2E2F303132333435363738393A3B3C3D3E3F404142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F606162636465666768696A6B6C6D6E6F707172737475767778797A7B7C7D7E7F808182838485868788898A8B8C8D8E8F909192939495969798999A9B9C9D9E9FA0A1A2A3A4A5A6A7A8A9AAABACADAEAFB0B1B2B3B4B5B6B7B8B9BABBBCBDBEBFC0C1C2C3C4C5C6C7C8C9CACBCCCDCECFD0D1D2D3D4D5D6D7D8D9DADBDCDDDEDFE0E1E2E3E4E5E6E7E8E9EAEBECEDEEEFF0F1F2F3F4F5F6F7F8F9FAFBFCFDFEFF',
+    x'000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F202122232425262728292A2B2C2D2E2F303132333435363738393A3B3C3D3E3F404142434445464748494A4B4C4D4E4F505152535455565758595A5B5C5D5E5F606162636465666768696A6B6C6D6E6F707172737475767778797A7B7C7D7E7F808182838485868788898A8B8C8D8E8F909192939495969798999A9B9C9D9E9FA0A1A2A3A4A5A6A7A8A9AAABACADAEAFB0B1B2B3B4B5B6B7B8B9BABBBCBDBEBFC0C1C2C3C4C5C6C7C8C9CACBCCCDCECFD0D1D2D3D4D5D6D7D8D9DADBDCDDDEDFE0E1E2E3E4E5E6E7E8E9EAEBECEDEEEFF0F1F2F3F4F5F6F7F8F9FAFBFCFDFEFF'
+);
```
Comment (hi-rustin) [inline tests/integration_tests/canal_json_adapter_compatibility/data/data.sql:157]:
```suggestion
);

```

**Review by hi-rustin — APPROVED** [review]

**Review by zhaoxinyu — APPROVED** [review]

File: tests/integration_tests/canal_json_adapter_compatibility/run.sh [hunk 2]
```diff
@@ -33,7 +33,8 @@ function run() {
 	run_sql_file $CUR/data/data.sql ${UP_TIDB_HOST} ${UP_TIDB_PORT}
 
 	# sync_diff can't check non-exist table, so we check expected tables are created in downstream first
-	check_table_exists test.multi_data_type ${DOWN_TIDB_HOST} ${DOWN_TIDB_PORT} 1000
+	check_table_exists test.binary_columns ${DOWN_TIDB_HOST} ${DOWN_TIDB_PORT} 1000
+
```
Comment (leoppro) [inline tests/integration_tests/canal_json_adapter_compatibility/run.sh:37]:
```suggestion
```

File: tests/integration_tests/canal_json_adapter_compatibility/run.sh [hunk 3]
```diff
@@ -33,7 +33,7 @@ function run() {
 	run_sql_file $CUR/data/data.sql ${UP_TIDB_HOST} ${UP_TIDB_PORT}
 
 	# sync_diff can't check non-exist table, so we check expected tables are created in downstream first
-	check_table_exists test.multi_data_type ${DOWN_TIDB_HOST} ${DOWN_TIDB_PORT} 1000
+	check_table_exists test.binary_columns ${DOWN_TIDB_HOST} ${DOWN_TIDB_PORT} 1000
```
Comment (hi-rustin) [inline tests/integration_tests/canal_json_adapter_compatibility/run.sh:36]:
```suggestion
	check_table_exists test.binary_columns ${DOWN_TIDB_HOST} ${DOWN_TIDB_PORT} 200
```

Comment (hi-rustin) [reply tests/integration_tests/canal_json_adapter_compatibility/run.sh:36]: Changing to 1000 is a mistake. Let's revert it and if there are still problems we need to contact the ci team to add pod resources.

**Comment by zhaoxinyu** [comment]
Unit test TestRegionWorkerExitWhenIsIdle failed.

**Pull request merged**